### PR TITLE
[WFLY-13675] Make dependency on PicketBox optional for Batch subsystem.

### PR DESCRIPTION
**Pull request by darranl**
https://issues.redhat.com/browse/WFLY-13675

### CODE REVIEW

File: batch-jberet/pom.xml [hunk 1]
```diff
@@ -75,6 +75,10 @@
             <groupId>org.jboss.msc</groupId>
             <artifactId>jboss-msc</artifactId>
         </dependency>
+        <dependency>
+            <groupId>org.picketbox</groupId>
+            <artifactId>picketbox</artifactId>
```
Comment (jamezp) [inline batch-jberet/pom.xml:80]: Should this have `<optional>true</optional>`?

Comment (darranl) [reply batch-jberet/pom.xml:80]: I suspect not as it will not compile without it - but maybe I should set the scope to provided to stop it leaking out.

Comment (jamezp) [reply batch-jberet/pom.xml:80]: I think optional is still valid as it's optional at runtime. If the legacy security is desired, then the dependency is not optional. However it should work at runtime without it.

**Comment by jamezp** [comment]
I'm not sure why those mixed domain tests are failing. They're passing for me locally so I kicked them off one more time.

**Comment by darranl** [comment]
I think those test failures are unrelated, they have just popped up on a load of jobs.

File: batch-jberet/pom.xml (hunk 1 above)
Comment (darranl) [reply batch-jberet/pom.xml:80]: TBH here we are talking about class loading for a subsystem, by that point the pom definition has been dropped and we have moved to module definitions for class loading.  I will add scope provided to all the picketbox dependencies as I add them so we can avoid transitive dependencies creeping in.  I will add optional if you really want it but as we don't use maven dependencies at runtime I think that is redundant information.

Comment (jamezp) [reply batch-jberet/pom.xml:80]: I've got no real strong opinion either way really. We do use `<optional>true</optional>` for the log tooling dependencies. I don't recall why we decided to do that though.

**Review by jamezp — APPROVED** [review]

**Pull request merged**